### ISSUE TITLE
Lists [NIP-51] Positive Interoperability Tracker

### ISSUE THREAD

**Issue by alltheseas** [nostrability/nostrability#82]
**app**|**Nip-51**|**kind30003**|**kind30000**|**kind10000*|**kind10003**|**kind 30015**|**kind 39089**|**comment**
-----|-----|-----|-----|-----|-----|-----|-----|-----
cornychat|✅ |.|.| | | |
damus|. |.|.| | | |
voyage|✅||✅|✅ | ✅|✅ |
gossip|✅|.|✅|✅ | ✅| ||30003 tbd; (private entries supported)
amethyst|✅|✅|✅| | | | |[v1.04.0](https://github.com/vitorpamplona/amethyst/releases/tag/v1.04.0) "Lists for all"; [PR#1169](https://github.com/vitorpamplona/amethyst/pull/1169) profile lists, [PR#1586](https://github.com/vitorpamplona/amethyst/pull/1586) bookmark lists
coracle|✅|✅|✅| ✅|. | | |kind 10000 mute via [release 0.6.19](https://github.com/coracle-social/coracle/releases/tag/0.6.19); kind 30015 not found
nostrudel|.|✅|.| | | | |bookmarks view, bookmark button for articles, mute list support
following.space|.|.|.| | | |✅| [kind-39089 nip-51 PR](https://github.com/nostr-protocol/nips/pull/1898); kind 30003 not relevant (focus on follow packs)
nostur|✅|✅|✅| | | | |[v1.20.0](https://github.com/nostur-com/nostur-ios-public/releases/tag/1.20.0) kind 30000 lists; [v1.13.0](https://github.com/nostur-com/nostur-ios-public/tree/1.13.0) bookmark categories; [v1.25.0](https://github.com/nostur-com/nostur-ios-public/releases/tag/1.25.0) bookmarks column
primal|~|~|.| | | | |[commit e354e417](https://github.com/PrimalHQ/primal-web-app/commit/e354e417) bookmarks multi-feed; no clear kind 30003/30000 NIP-51 impl
notedeck|p|||||||v0.8.0-rc2 'Follows view and follows list columns'
jumble|✅|✅|✅|||||Bookmarks, mute lists, pin lists (3 providers)
njump|✅||✅|||||Kind 30000 FollowSet + kind 39089 StarterPack with dedicated templates
yakihonne|✅|||||||na-gro audit: curation management
snort|✅|✅|✅|✅||||na-gro audit: comprehensive NIP-51
lume|✅||✅|||||na-gro audit: desktop client
nostria|✅||✅|||||na-gro audit: list management system
habla.news|✅||✅|||||na-gro audit: bookmark/mute lists, state atoms
0xchat|✅|||||||na-gro audit: mobile chat
flotilla|✅|||||||na-gro audit: community client
chachi|✅|||||||na-gro audit: group client
zap.stream|✅|||||||na-gro audit: via @snort/system
shopstr|✅|||||||na-gro audit: marketplace

**Comment by staab** [comment]
Coracle reads many types of lists in order to incorporate them into feeds, but in particular I'm maintaining backward compatibility for kind 30003 as the old version of custom feeds (deprecated, includes a migration path), and allowing kind 30000 list curation.

**Comment by alltheseas** [comment]
@staab does coracle allow for 

private contact lists?
private mute lists?
private bookmarks?

**Comment by staab** [comment]
No, currently everything is public (I do leave encrypted content alone)

**Comment by alltheseas** [comment]
@vicariousdrama can you specify nip-51 kind support in cornychat, and other nip-51 relevant details

**Comment by vicariousdrama** [comment]
> @vicariousdrama can you specify nip-51 kind support in cornychat, and other nip-51 relevant details

Corny Chat currently supports two kinds of lists. Note that while these are named as `Sets`, this is only for consistency with the NIP-51 nomenclature. It's important to note that these truly are lists, not sets when it comes to the ability to have duplicate entries, and order of `r` tags matters.

## kind 30388: Slide Set

A slide set is a collection of URLs that represent images that can be displayed within a Corny Chat room.  Owners and moderators with a nostr  extension can export and import slide sets from room settings.  As a replaceable event, a `d` tag is required to be set, and is defaulted by the application based on a concatenation of the string `cornychat`, the room identifier, and the date and time in ISO format.  Reusing the same value allows for replacing a prior published event when doing an export.  The expected tags for import, which are set automatically during export, are as follows:
```
tags: [
   ["d", "the-d-tag"],
   ["name", "User provided name of the slide set"],
   ["about", "User provided description of the slide set"],
   ["image", "URI selectable from user of an image in the slide set"],
   ["L", "com.cornychat"],
   ["l", "Corny Chat Slide Set", "com.cornychat"],
   ["r", "URI of image", "optional caption of image"],     // repeated
]
```
Other applications may make Slide Sets or use the same kind 30388 as well. But they will not show in the list of slide sets to import unless they include the referenced label namespace and label value.  Slide sets will also be excluded if using the `expiration` tag and such timestamp has passed.

## kind 31388: Link Set

A link set is a collection of URLs that represent captioned hyperlinks or groupings of links which can be displayed within the links resources area of a Corny Chat room.  Owners and moderators with a nostr extension can export and import link sets from room settings.  As a replaceable event, a `d` tag is required to be set, and is defaulted by the application based on a concatenation of the string `cornychat`, the room identifier, and the date and time in ISO format.  Reusing the same
 value allows for replacing a prior published event when doing an export.  The expected tags for import, which are set automatically during export are as follows:
```
tags: [
   ["d", "the-d-tag"],
   ["name", "User provided name of the link set"],
   ["about", "User provided description of the link set"],
   ["image", "URI of an image denoting it is a link set"],
   ["L", "com.cornychat"],
   ["l", "Corny Chat Link Set", "com.cornychat"],
   ["r", "optional URI of link", "optional caption for link or group without a link"],     // repeated
]
```
The order of `r` tags is set by users of the application and this order should be maintained only changed within an interface allowing the user to reorder. Other applications may make Link Sets or use the same kind 31388 as well. But they will not show in the list of link sets to import unless they include the referenced label namespace and label value.  Link sets will also be excluded if using the `expiration` tag and such timestamp has passed.

_Future plans include support for Favorite Rooms (32388), Folders (33388 unless defined otherwise through changes to NIP-51), Follow Sets (30000), and Emoji Sets (30030)_

**Comment by dluvian** [comment]
Voyage reads and writes:
- `kind:10000` Mute List
- `kind:10003` Bookmarks
- `kind:10015` Interests
- `kind:30000` Follow Sets
- `kind:30015` Interest Sets

**Comment by dluvian** [comment]
Last time I checked Amethyst puts everything in the contact list (deprecated) but is able to read `kind:30000`

**Comment by alltheseas** [comment]
Following.space

Example create list event


`{
  "created_at": 1746194302,
  "content": "",
  "tags": [
    [
      "title",
      "Human Architecture & Local Vernacular"
    ],
    [
      "d",
      "y156932o9xfh"
    ],
    [
      "image",
      "https://c4.wallpaperflare.com/wallpaper/82/123/783/architecture-building-bulgaria-village-wallpaper-preview.jpg"
    ],
    [
      "p",
      "5db2be23cde61dd0a69e667a021a943fa38760104fe4160ce13b1a097e9fe447"
    ],
    [
      "description",
      "Aesthetic architecture, craftsmanship, and localism"
    ]
  ],
  "kind": 39089,
  "pubkey": "17538dc2a62769d09443f18c37cbe358fab5bbf981173542aa7c5ff171ed77c4",
  "id": "6c237670913b3886a81596e0e876e7e023c5435f7685d9eef4489db139683786",
  "sig": "cb547df02f0971b24b28c96490bd310d88fbce7ce67b3f5ed3b8ca90d27b29541ced2df255481022bb119841f7370d2a43f68deb6c2af104353756e97c1b9d09"
}

kind-39089 NIP-51 PR: https://github.com/nostr-protocol/nips/pull/1898

**Comment by alltheseas** [comment]
Following.space utilizes  "kind": 39089,
 
kind-39089 NIP-51 PR: https://github.com/nostr-protocol/nips/pull/1898

**Comment by alltheseas** [comment]
Amethyst is switchin from nip-04 to nip-44 encryption of lists https://github.com/nostrability/nostrability/issues/243
